signer/src/storage/model.rs

```diff
@@ -1004,6 +1011,52 @@ impl PartialOrd for StacksPrincipal {
 #[derive(Debug, Clone, Hash, PartialEq, Eq, PartialOrd, Ord)]
 pub struct ScriptPubKey(bitcoin::ScriptBuf);
 
+/// A taproot script hash.
+#[derive(Debug, Clone, Hash, PartialEq, Eq, PartialOrd, Ord)]
+pub struct TaprootScriptHash(bitcoin::TapNodeHash);
+
+impl Deref for TaprootScriptHash {
+    type Target = bitcoin::TapNodeHash;
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl From<bitcoin::TapNodeHash> for TaprootScriptHash {
+    fn from(value: bitcoin::TapNodeHash) -> Self {
+        Self(value)
+    }
+}
+
+impl Default for TaprootScriptHash {
+    fn default() -> Self {
+        TaprootScriptHash::from_byte_array([0; 32])
+    }
+}
+
+impl TaprootScriptHash {
+    /// Create a new taproot script hash with all zeroes
+    pub fn new() -> Self {
+        Default::default()
+    }
+    /// Return the inner bytes for the taproot script hash
+    pub fn into_bytes(&self) -> [u8; 32] {
+        self.0.to_byte_array()
+    }
+
+    /// Convert a byte array into a taproot script hash.
+    pub fn from_byte_array(bytes: [u8; 32]) -> Self {
+        bitcoin::TapNodeHash::from_byte_array(bytes).into()
+    }
+
+    /// Compute a taproot script hash from a script pubkey.
+    pub fn from_script_pubkey(script_pubkey: &ScriptPubKey) -> Self {
+        // TODO: I have no idea what this LeafVersion means atm
```

Comment:
I think I used correct LeafVersion here but somebody confirm pls

Reply:
This is the correct `LeafVersion` version. Also, right now, the alternative here is the taproot annex version, which is essentially unused at the moment (I think it may be considered non-standard by default or something like that).